parser/parser_test.go

```diff
@@ -1412,6 +1412,24 @@ func (s *testParserSuite) TestPrivilege(c *C) {
 	table := []testCase{
 		// for create user
 		{`CREATE USER 'test'`, true},
+		{`CREATE USER test`, true},
+		{"CREATE USER `test`", true},
+		{"CREATE USER test-user", false},
+		{"CREATE USER test.user", false},
+		{"CREATE USER 'test-user'", true},
+		{"CREATE USER `test-user`", true},
+		{"CREATE USER test.user", false},
+		{"CREATE USER 'test.user'", true},
+		{"CREATE USER `test.user`", true},
+		{"CREATE USER uesr1@localhost", true},
```

Comment:
It is copied from the original issue https://github.com/pingcap/tidb/issues/3693